docs/docs/concepts/flows.mdx

```diff
@@ -460,23 +460,44 @@ If the "final_confirmation" slot is already filled, the bot will clear it, ask
 the `collect_information` step and fill the slot again. This ensures the confirmation
 `collect_information` step is always asked.
 
-#### Scopes
+#### Reset slot value after flow execution
 
-Scopes determine the lifespan of a slot or variable. There are two types of
-scopes available:
+You can determine whether to reset the slot value after the flow execution or not by setting the optional
+`reset_after_flow_ends` property to `true` or `false` respectively. If the `reset_after_flow_ends` property is set to
+`True`, the slot value exists only for the duration of the flow. Once the flow ends, the slot value is either reset to
+`null` or the slot's initial value (if provided in the domain file under `slots`). If instead the `reset_after_flow_ends`
+property is set to `false`, this indicates that the slot value should persist across flows until it is explicitly changed.
 
-- **flow**: The variable exists only for the duration of the flow. Once the flow
-  ends, the variable ceases to exist.
-- **global**: The variable persists across flows until it is explicitly changed.
-
-The slot can also be configured to be cleared at the end of the flow execution.
 The following configuration would clear the answer to the `collect_information` step when a
 flow is completed.
 
 ```yaml
 - id: "another_confirmation_step"
-  collect_information: "yes_or_no" # slot name
-  scope: "flow"
+  collect_information: "my_custom_slot" # slot name
+  reset_after_flow_ends: true
```

Comment:
product q: do we have the same option under `set_slots`?

Reply:
Seems not, I just checked the [initial implementation](https://github.com/RasaHQ/rasa/blob/4fbb0ea96162ee3bc33be5a0335840a433d1e614/rasa/shared/core/flows/flow.py#L1058) and no mention of this option for this step type. Also the [PRD ](https://www.notion.so/rasa/PRD-Flow-spec-Studio-034f6810b3254ed09fe4e3a2c6a4519c?pvs=4#0d6f14fc3de34b8a8081d1973eb44d02) hints at a setting that indicates whether to reset or not, but nothing explicit, so it might have been overlooked.

Reply:
can you check this week to see if this is indeed needed or not?

Reply:
Yep, already sent a question on Slack to both PMs.

Reply:
[It's been decided ](https://rasa-hq.slack.com/archives/C047U7ZBKM1/p1695795629580879?thread_ts=1695632776.236379&cid=C047U7ZBKM1) that we won't make this property user facing for `set_slots`, but we will need to make sure that by default for now the slots from a `set_slots` step will be reset at the end of a flow. A new ticket was added to Atom board for next sprint, I can tackle any docs related updates as part of that.